### ISSUE TITLE
Expand tS api to support tA in other languages

### ISSUE THREAD

**Issue by neutrinog** [unfoldingWord-dev/d43-catalog#57]
tA has been translated to [Croatian](https://git.door43.org/Door43-Catalog/hr_ta) and users would like to access it inside of tS.

Right now only english is processed and uploaded to the tS v2 api.

Example:
* `en` https://api.unfoldingword.org/ta/txt/1/en/audio_2.json
* but we also want `hr` https://api.unfoldingword.org/ta/txt/1/hr/audio_2.json (broken)

Once the api has been updated we'll need to improve the [android client](https://github.com/unfoldingWord-dev/android-door43-client/blob/master/door43-client/src/main/java/org/unfoldingword/door43client/LegacyTools.java#L112) and [node client](https://github.com/unfoldingWord-dev/node-door43-client/blob/master/lib/main.js#L463) to also support other languages. Both are currently hard coded to English.

Finally, the Android and desktop apps will need to be updated to display the correct translation of tA. This should depend on the selected source translation in the project (Android already does this).

Here are some key locations in the code:
* [Android - ReviewModeFragment](https://github.com/unfoldingWord-dev/ts-android/blob/develop/app/src/main/java/com/door43/translationstudio/ui/translate/ReviewModeFragment.java#L261)
* [Desktop - createAcademyWindow](https://github.com/unfoldingWord-dev/ts-desktop/blob/develop/main.js#L257) (we'll need to pass in the language to load)
* [Desktop - ts-academy](https://github.com/unfoldingWord-dev/ts-desktop/blob/develop/src/elements/ts-main/ts-academy.html#L105)

**Comment by neutrinog** [comment]
I've determined that tA is not processed at all by the ts v2 api. The english tW was manually uploaded in 2017.

**Comment by jag3773** [comment]
As a result, we've decided not to build this feature into the deprecated tS v2 API.  Support for tA is available in the v3 API and apps that want to access it can use it there.

This does mean that tS v12 won't support viewing tA in other languages.

**Comment by jag3773** [comment]
I'd suggest that we don't touch the catalog but rather have tS hit https://api.door43.org/v3/subjects/Translation_Academy.json directly.  Since there isn't any code in tS to handle tA right now anyway, makes sense to just add the code there to get it from the latest published location.

**Comment by neutrinog** [comment]
Originally, all of the api content was downloaded and managed from a separate library.
Because translationAcademy will be coming from a different api end point and format, we'll need to provide a new way to view, update, and download translationAcademy translations. 

@jag3773 can you provide some feedback on the below approach?

* The old translationAcademy content that was used in the app will be deprecated.
* When opening translationAcademy from a link in a project it will automatically use the same language as the source text.
  * If an update is available for the translationAcademy translation, the user will be prompted to download it.
* When opening translationAcademy from the menu, a dialog will first appear that asks the user to choose a language.
  * If a translation is available but not downloaded, it can be downloaded by clicking on a download button.
  * If a translation update is available, it can be downloaded as well.
* When the user updates the list of available source texts, the translationAcademy catalog will be downloaded from https://api.door43.org/v3/subjects/Translation_Academy.json.

> Concerns: Will any of the tA links within tN become invalid because we are using tA from the v3 api? There have been some structural changes to tA since the v2 days.

**Comment by jag3773** [comment]
@neutrinog That approach sounds like a good way to go.

Regarding your concerns, I don't think that will be a problem, especially because the tN content *is being updated* while the tA content has not been.  In other words, having the latest tA available is likely to fix areas where there might have been a link mismatch.  However, in the vast majority of cases we have tried to not change the paths/links to any of the articles.

Some articles have been added and a couple of the manuals were removed, but the removed manuals were the ones not linked to from tN anyway (tech and audio).

Note that @klappy has some code in https://github.com/unfoldingWord-box3/translation-helps for showing Markdown as HTML including rendering the RC links.  If there is a straightforward way to use that then great :)

**Comment by klappy** [comment]
There are plans in the not so distant future to abstract that out as it’s own library on npm.

**Comment by neutrinog** [comment]
@jag3773 yes, but tN is going through the tS api. I'll need to double check that we aren't doctoring any of the links before it gets to tS.

**Comment by neutrinog** [comment]
Just about finished. There's a few usability things I need to add including:

* if there is an update to the tA translation the user should be prompted to "update" it or "continue". Right now it looks like a normal download.
* Display the tA language in the confirmation and download dialogs.
* Download the tA catalog when performing regular application updates.
* Pre-cache the catalog and several versions of tA in the bundled app.

Checking for updates
![checking-for-updates](https://user-images.githubusercontent.com/166412/65872375-9d07f280-e3aa-11e9-8184-92bda7189d8b.gif)

Downloading tA
![downloading-ta](https://user-images.githubusercontent.com/166412/65872393-a5602d80-e3aa-11e9-9021-9f5f2f59abb5.gif)

Opening tA from a contextual link
![opening-contextual-link-with-downloaded-ta](https://user-images.githubusercontent.com/166412/65872407-af822c00-e3aa-11e9-9348-318f92d42d2b.gif)

Opening tA from a contextual link when the translation of tA has not yet been downloaded.
![opening-contextual-link-without-downloaded-ta](https://user-images.githubusercontent.com/166412/65872439-bb6dee00-e3aa-11e9-8b07-2270db932402.gif)

**Comment by jag3773** [comment]
Looking really good @neutrinog ! One thing that I want to make sure that we preserve is fall back to English tA if tA is not available in the GL.  Also, if the user doesn't want to download the GL version, then English should be shown.

**Comment by neutrinog** [comment]
@jag3773 what versions of tA should be bundled with the app?

**Comment by jag3773** [comment]
Let's do all that are published.
